code/modules/mob/living/basic/basic.dm

```diff
@@ -35,6 +35,10 @@
 	var/attack_vis_effect
 	///Played when someone punches the creature.
 	var/attacked_sound = SFX_PUNCH //This should be an element
+	/// How often can you melee attack?
+	var/melee_attack_cooldown = 2 SECONDS
+	/// Timer tracking when we can next perform a melee attack
+	COOLDOWN_DECLARE(melee_attack_cooldown_timer)
```

Comment:
We already have a timer for determining when we can perform a melee attack, it's Nextmove.............

AI trees would already abide by it if they didn't skip the whole chain and run melee_attack................

Reply:
My reasoning for not using `nextMove` is that it blocks clicks generally and that feels annoying when it's like two seconds long, however on reflection I don't know what else you're going to be doing with your nextmove as basically any mob. You're not likely to stop to pet Ian between these.
I have overcomplicated this.